core/src/main/java/org/bitcoinj/core/AbstractAddress.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2018 Andreas Schildbach
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.bitcoinj.core;
+
+import javax.annotation.Nullable;
+
+public class AbstractAddress extends VersionedChecksummedBytes {
```

Comment:
Could we rename this to GenericAddress?

Reply:
My idea was to rename it to Address, so it can be a plug-in replacement for the current Address (which in turn I would rename to LegacyAddress).

Reply:
Renaming to Address is a much cleaner approach.

I forgot to ask about adding a method from `AddressScript`:
`public abstract Script.ScriptType getScriptType();`

Reply:
Added.